docker/resource_docker_plugin.go

```diff
@@ -0,0 +1,67 @@
+package docker
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func resourceDockerPlugin() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceDockerPluginCreate,
+		Read:   resourceDockerPluginRead,
+		Update: resourceDockerPluginUpdate,
+		Delete: resourceDockerPluginDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"plugin_reference": {
+				Type:        schema.TypeString,
+				Description: "Docker Plugin Reference.",
+				Required:    true,
+				ForceNew:    true,
+			},
+			"alias": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "Docker Plugin alias.",
+			},
+			"disabled": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+			"grant_all_permissions": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "If true, grant all permissions necessary to run the plugin",
+			},
+			"env": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+
+			"disable_when_set": {
```

Comment:
I want a better attribute name.

Reply:
To run `docker plugin set`, the plugin must be disabled.

ex.

```
$ docker plugin set 27784976e147 DEBUG=2
Error response from daemon: cannot set on an active plugin, disable plugin before setting
```

Reply:
what is the behavior when disable_when_set is false?

Reply:
I just had a look, you should remove this attribute and just disable the plugin on all updates.

Reply:
When we try to update a active plugin

1. If disable_when_set is false (default), the update would fail
1. If disable_when_set is true, the plugin becomes disable temporarily

I introduce this attribute because I concern someone may prefer the failure rather than temporal disable.

But I think removing this attribute is possibility too because it is simpler.

Reply:
Terraforms workflow shouldn't fail when updating a resource. There would be no way in terraform to explicitly disable, update, and reenable. I think it would be best to remove this attribute.

Reply:
This attribute is like a force option.
Removing this attribute is like forcibly update of the plugin always.
I wonder if that is better or not.
I'm not familiar with Docker plugin, so I don't know whether it has no problem to disable the plugin temporarily.

@mavogel What do you think?

Reply:
After reading the docs, I'd also suggest removing the attribute. As a workflow, folks you have to explicitly disable (enable=false) a plugin, update to a new version and then re-enable. If we see later that it's better/desired to use with e.g. `auto_update=true`, we could implement this so the workflow of disable, update and re-enable is done in a function. IMHO.

Reply:
Thank you for your reivew.
I see.
Updated. abec1d568c9dae0066d4a7268e74747c999c6c8f 17d8fc6651c17640a7e259885833ecc565678682